api/src/lib/sentry/index.ts

```diff
@@ -0,0 +1,60 @@
+import * as path from 'path';
+import Raven from 'raven';
+import { errResponse } from '../utils/response';
+
+const parseUser = (event: AWSLambda.APIGatewayProxyEvent): SentryUserInfo => {
+  if (!event.requestContext.authorizer) { return; }
+
+  const authorizerUser = JSON.parse(event.requestContext.authorizer.user);
+  return {
+    email: authorizerUser.email,
+    username: authorizerUser.username
+  };
+};
+
+const getSourceMaps: Raven.DataCallback = (data: {[key: string]: any}): {[key: string]: any} => {
+  const stacktrace: any = data.exception && data.exception[0].stacktrace;
+  if (stacktrace && stacktrace.frames) {
+    stacktrace.frames.forEach((frame: any) => {
+      const p = path.relative('/var/task/', frame.filename);
+      frame.filename = 'app:///' + p.split('/')[0] + '/' + p;
+    });
+  }
+
+  return data;
+};
+
+export default class SentryWrapper {
+  public static handler(lambdaHandler: LambdaHandler): LambdaHandler {
+    return async (event: AWSLambda.APIGatewayProxyEvent): Promise<AWSLambda.APIGatewayProxyResult> => {
+      try {
+        return await lambdaHandler(event);  
+      } catch (error) {
+        if (!process.env.SENTRY_DSN) {
+          console.error('SENTRY_DSN not found in env. Throwing error.');
+          throw error;
```

Comment:
I added this so you can run the handler locally without logging any errors to Sentry during development and still get to see the error. If you run `Raven.captureException` wihout `SENTRY_DSN` configured it will quietly do nothing. The error as caught, but it's not reported to Sentry so we'll never know about it. 

For the same reason, we should never deploy without `SENTRY_DSN` being configured. I added an `AllowedPattern` requirement to the `SentryDsn` parameter in the SAM template. That means it should never deploy without the `SENTRY_DSN` configured.